sdk/core/Azure.Core/src/Shared/OperationPollingStrategy.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Strategy to control polling behavior of LRO operation.
+    /// </summary>
+    internal abstract class OperationPollingStrategy
+    {
+        protected static readonly OperationPollingStrategy DefaultPollingStrategy = new ConstantPollingStrategy();
+
+        /// <summary>
+        /// Factory method to choose a proper <see cref="OperationPollingStrategy"/> based on given <paramref name="response"/> and <paramref name="customPollingStrategy"/>.
+        /// </summary>
+        /// <param name="response"> Server response. Normally it's the initial response of an LRO operation. </param>
+        /// <param name="customPollingStrategy"> Default <see cref="OperationPollingStrategy"/>. </param>
+        /// <returns> A <see cref="RetryAfterPollingStrategy"/> if there is a valid retry-after heawder in given <paramref name="response"/>; <paramref name="customPollingStrategy"/> otherwise. </returns>
+        public static OperationPollingStrategy ChoosePollingStrategy(Response response, OperationPollingStrategy? customPollingStrategy)
+        {
+            return RetryAfterPollingStrategy.IsRetryAfterPresent(response) ? new RetryAfterPollingStrategy(response) : customPollingStrategy ?? DefaultPollingStrategy;
```

Comment:
Base class shouldn't reference one of its implementation. I'd prefer each strategy to have `TryGetNextWait(Response response, TimeSpan? intervalHint, out TimeSpan interval)` method, so that list of strategies can be iterated